From 7b9e8490d3a1f86530b0cd405e12be02d4d68a5d Mon Sep 17 00:00:00 2001
From: rmp22 <195054967+rmp22@users.noreply.github.com>
Date: Thu, 6 Feb 2025 06:15:21 +0800
Subject: [PATCH 43/74] BroadcastConstants: Reduce max running broadcast limits

Signed-off-by: rmp22 <195054967+rmp22@users.noreply.github.com>
---
 .../android/server/am/BroadcastConstants.java | 21 +++++++------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/services/core/java/com/android/server/am/BroadcastConstants.java b/services/core/java/com/android/server/am/BroadcastConstants.java
index fe4fa1068bb3..86957ec81ec2 100644
--- a/services/core/java/com/android/server/am/BroadcastConstants.java
+++ b/services/core/java/com/android/server/am/BroadcastConstants.java
@@ -117,8 +117,7 @@ public class BroadcastConstants {
      */
     public int MAX_RUNNING_PROCESS_QUEUES = DEFAULT_MAX_RUNNING_PROCESS_QUEUES;
     private static final String KEY_MAX_RUNNING_PROCESS_QUEUES = "bcast_max_running_process_queues";
-    private static final int DEFAULT_MAX_RUNNING_PROCESS_QUEUES =
-            ActivityManager.isLowRamDeviceStatic() ? 2 : 4;
+    private static final int DEFAULT_MAX_RUNNING_PROCESS_QUEUES = 2;
 
     /**
      * For {@link BroadcastQueueImpl}: Additional running process queue parallelism beyond
@@ -157,8 +156,7 @@ public class BroadcastConstants {
     public int MAX_RUNNING_ACTIVE_BROADCASTS = DEFAULT_MAX_RUNNING_ACTIVE_BROADCASTS;
     private static final String KEY_MAX_RUNNING_ACTIVE_BROADCASTS =
             "bcast_max_running_active_broadcasts";
-    private static final int DEFAULT_MAX_RUNNING_ACTIVE_BROADCASTS =
-            ActivityManager.isLowRamDeviceStatic() ? 8 : 16;
+    private static final int DEFAULT_MAX_RUNNING_ACTIVE_BROADCASTS = 8;
 
     /**
      * For {@link BroadcastQueueImpl}: Maximum number of active "blocking" broadcasts
@@ -170,8 +168,7 @@ public class BroadcastConstants {
     public int MAX_CORE_RUNNING_BLOCKING_BROADCASTS = DEFAULT_MAX_CORE_RUNNING_BLOCKING_BROADCASTS;
     private static final String KEY_CORE_MAX_RUNNING_BLOCKING_BROADCASTS =
             "bcast_max_core_running_blocking_broadcasts";
-    private static final int DEFAULT_MAX_CORE_RUNNING_BLOCKING_BROADCASTS =
-            ActivityManager.isLowRamDeviceStatic() ? 8 : 16;
+    private static final int DEFAULT_MAX_CORE_RUNNING_BLOCKING_BROADCASTS = 8;
 
     /**
      * For {@link BroadcastQueueImpl}: Maximum number of active non-"blocking" broadcasts
@@ -184,8 +181,7 @@ public class BroadcastConstants {
             DEFAULT_MAX_CORE_RUNNING_NON_BLOCKING_BROADCASTS;
     private static final String KEY_CORE_MAX_RUNNING_NON_BLOCKING_BROADCASTS =
             "bcast_max_core_running_non_blocking_broadcasts";
-    private static final int DEFAULT_MAX_CORE_RUNNING_NON_BLOCKING_BROADCASTS =
-            ActivityManager.isLowRamDeviceStatic() ? 32 : 64;
+    private static final int DEFAULT_MAX_CORE_RUNNING_NON_BLOCKING_BROADCASTS = 32;
 
     /**
      * For {@link BroadcastQueueImpl}: Maximum number of pending
@@ -194,8 +190,7 @@ public class BroadcastConstants {
      */
     public int MAX_PENDING_BROADCASTS = DEFAULT_MAX_PENDING_BROADCASTS;
     private static final String KEY_MAX_PENDING_BROADCASTS = "bcast_max_pending_broadcasts";
-    private static final int DEFAULT_MAX_PENDING_BROADCASTS =
-            ActivityManager.isLowRamDeviceStatic() ? 128 : 256;
+    private static final int DEFAULT_MAX_PENDING_BROADCASTS = 128;
 
     /**
      * For {@link BroadcastQueueImpl}: Delay to apply to normal
@@ -248,8 +243,7 @@ public class BroadcastConstants {
      */
     public int MAX_HISTORY_COMPLETE_SIZE = DEFAULT_MAX_HISTORY_COMPLETE_SIZE;
     private static final String KEY_MAX_HISTORY_COMPLETE_SIZE = "bcast_max_history_complete_size";
-    private static final int DEFAULT_MAX_HISTORY_COMPLETE_SIZE =
-            ActivityManager.isLowRamDeviceStatic() ? 64 : 256;
+    private static final int DEFAULT_MAX_HISTORY_COMPLETE_SIZE = 64;
 
     /**
      * For {@link BroadcastQueueImpl}: Maximum number of summarized
@@ -257,8 +251,7 @@ public class BroadcastConstants {
      */
     public int MAX_HISTORY_SUMMARY_SIZE = DEFAULT_MAX_HISTORY_SUMMARY_SIZE;
     private static final String KEY_MAX_HISTORY_SUMMARY_SIZE = "bcast_max_history_summary_size";
-    private static final int DEFAULT_MAX_HISTORY_SUMMARY_SIZE =
-            ActivityManager.isLowRamDeviceStatic() ? 256 : 1024;
+    private static final int DEFAULT_MAX_HISTORY_SUMMARY_SIZE = 256;
 
     /**
      * For {@link BroadcastRecord}: Default to treating all broadcasts sent by
-- 
2.43.0

